### ISSUE TITLE
Support for arbitrary value in AssemblyInformationalVersionAttribute

### ISSUE THREAD

**Issue by tmat** [Microsoft/visualfsharp#4822]
The compiler reports a warning when the value of AssemblyInformationalVersionAttribute is not a valid version string.

.NET SDK 2.1.300-rc1-008673

Repro:
```
> dotnet new classlib --language F#
```

Set value in the project:
```xml
<InformationalVersion>1.0.0+a1686e12e2b5dbbf0773de915f8b6039041dd4c1</InformationalVersion>
```

```
> dotnet build
...
FSC : warning FS2003: An System.Reflection.AssemblyInformationalVersionAttribute specified version '1.0.0+a1686e12e2b5dbbf0773de915f8b6039041dd4c1', but this value is invalid and has been ignored 
```

The C# and VB compilers accept arbitrary values.

SourceLink package adds git commit sha to the version (format above).

**Comment by tmat** [comment]
@KevinRansom

**Comment by KevinRansom** [comment]
I will take care of it.

**Comment by cartermp** [comment]
Link #5336

**Comment by ctaggart** [comment]
Even though this was merged on July 18th to the master branch, the fix has not made it to a stable .NET Core SDK yet. It still fails on the most recent stable .NET Core SDK 2.1.402. Any idea when it will make it to a  stable released version? I just came across this in the Giraffe build when trying to enable SourceLink there https://github.com/giraffe-fsharp/Giraffe/pull/304. They have `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>`, so it failed the build.

Workaround is https://github.com/giraffe-fsharp/Giraffe/blob/08bd0cbc802f1d9eeae460126ab481ac9783f9f8/src/Directory.Build.props#L7-L8 
``` xml
    <!-- Support for arbitrary value in AssemblyInformationalVersionAttribute https://github.com/Microsoft/visualfsharp/issues/4822 -->
    <NoWarn>FS2003</NoWarn>
```
